timelock/program/src/state/timelock_config.rs

```diff
@@ -0,0 +1,223 @@
+use super::enums::{ConsensusAlgorithm, ExecutionType, TimelockType, VotingEntryRule};
+use super::UNINITIALIZED_VERSION;
+use arrayref::{array_mut_ref, array_ref, array_refs, mut_array_refs};
+use solana_program::{
+    program_error::ProgramError,
+    program_pack::{IsInitialized, Pack, Sealed},
+    pubkey::Pubkey,
+};
+
+/// STRUCT VERSION
+pub const TIMELOCK_CONFIG_VERSION: u8 = 1;
+/// max name length
+pub const CONFIG_NAME_LENGTH: usize = 32;
+/// Timelock Config
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct TimelockConfig {
+    ///version
+    pub version: u8,
+    /// Consensus Algorithm
+    pub consensus_algorithm: ConsensusAlgorithm,
+    /// Execution type
+    pub execution_type: ExecutionType,
+    /// Timelock Type
+    pub timelock_type: TimelockType,
+    /// Voting entry rule
+    pub voting_entry_rule: VotingEntryRule,
+    /// Minimum slot time-distance from creation of proposal for an instruction to be placed
+    pub minimum_slot_waiting_period: u64,
+    /// Governance mint
+    pub governance_mint: Pubkey,
+    /// Council mint
+    pub council_mint: Pubkey,
+    /// Program ID that is tied to this config (optional)
```

Comment:
Yeah when I wrote this I hadn't learned that trick yet ;) I will make a note.

Reply:
Looking at this some more, would it be best to always have a program id?  Even for a multi-program timelock config, it may make sense to have one primary program id used to generate program addresses

Reply:
Yeah, I'm not averse to it. Let's just require it, it makes everything else a bit easier. It also shouldn't be difficult for people to just spin up a holding program. They could use an empty account for all we care, it's not as if I check for executable, or even that it exists!

Reply:
yeah true, sounds good!